packages/core/components/Modal/CodeSnippet/ConvertToZarr.tsx

```diff
@@ -0,0 +1,307 @@
+
+type UIOpts = {
+    destination: string;
+    scenesMode: "all" | "single";
+    sceneIndex: string;
+};
+
+function Snippet({
+    title,
+    text,
+    copyAria,
+}: {
+    title: "Setup" | "Code";
+    text?: string;
+    copyAria: string;
+}) {
+    const [copied, setCopied] = React.useState(false);
+    const value = text ?? "";
+
+    const onCopy = () => {
+        if (!value) return;
+        navigator.clipboard.writeText(value);
+        setCopied(true);
+        window.setTimeout(() => setCopied(false), 900);
+    };
+
+    return (
+        <>
+            <div className={styles.snipHeader}>
+                <div className={styles.snipTitle}>{title}</div>
+                <TooltipHost
+                    content={copied ? "Copied!" : undefined}
+                    directionalHint={DirectionalHint.leftCenter}
+                >
+                    <TertiaryButton
+                        iconName="Copy"
+                        aria-label={copyAria}
+                        title="Copy"
+                        className={styles.snipCopyBtn}
+                        onClick={onCopy}
+                    />
+                </TooltipHost>
+            </div>
+
+            <pre className={styles.codeBox}>
+                <code>{value || (title === "Setup" ? "# Install..." : "Code...")}</code>
+            </pre>
+        </>
+    );
+}
+
+export default function ConvertToZarr({ onDismiss }: ModalProps) {
+    const dispatch = useDispatch();
+    const fileSelection = useSelector(selection.selectors.getFileSelection);
+    const snippet = useSelector(interaction.selectors.getConvertFilesSnippet);
+
+    // Defaults
+    const [opts, setOpts] = React.useState<UIOpts>({
+        destination: "",
+        scenesMode: "all",
+        sceneIndex: "0",
+    });
+
+    const safe = (s: string) =>
+        String(s ?? "")
+            .replace(/\\/g, "\\\\")
+            .replace(/"/g, '\\"');
+
+    const toScenesPy = (mode: UIOpts["scenesMode"], idx: string): string | null => {
+        if (mode !== "single") return null; // omit when "all"
+        const v = (idx || "").trim();
+        if (!v) return null;
+        if (/^\d+$/.test(v)) return String(parseInt(v, 10));
+        return null;
+    };
+
+    // Build DEFAULTS dict string with only present keys
+    const buildDefaultsPy = (ui: UIOpts) => {
+        const lines: string[] = [];
+
+        const dest = (ui.destination || "").trim();
+        if (dest) {
+            lines.push(`"destination": ${JSON.stringify(dest)},`);
+        }
+
+        const scenesPy = toScenesPy(ui.scenesMode, ui.sceneIndex);
+        if (scenesPy !== null) {
+            lines.push(`"scenes": ${scenesPy},`);
+        }
+
+        if (!lines.length) return "DEFAULTS = {}";
+        return `DEFAULTS = {\n${lines.map((l) => `    ${l}`).join("\n")}\n}`;
+    };
+
+    const detailsRef = React.useRef<Array<{ path: string }>>([]);
+
+    const regenerate = React.useCallback(
```

Comment:
function name isn't super clear to me, could maybe benefit from brief explanatory comment